components/brave_vpn/winvpntool.cc

```diff
@@ -0,0 +1,712 @@
+          wprintf(L"\n\t\tdwSize=%d", lpRasSubEntry->dwSize);
+          wprintf(L"\n\t\tdwfFlags=%d", lpRasSubEntry->dwfFlags);
+          wprintf(L"\n\t\tszDeviceType=%s", lpRasSubEntry->szDeviceType);
+          wprintf(L"\n\t\tszDeviceName=%s", lpRasSubEntry->szDeviceName);
+          wprintf(L"\n\t\tszLocalPhoneNumber=%s",
+                  lpRasSubEntry->szLocalPhoneNumber);
+          wprintf(L"\n\t\tdwAlternateOffset=%d",
+                  lpRasSubEntry->dwAlternateOffset);
+          HeapFree(GetProcessHeap(), 0, lpRasSubEntry);
+          lpRasSubEntry = NULL;
+        } else {
+          wprintf(L"\n\tError calling RasGetSubEntryProperties: ");
+          PrintRasError(dwRet);
+        }
+      }
+    }
+
+    wprintf(L"\n");
+    // Deallocate memory for the entry buffer
+    HeapFree(GetProcessHeap(), 0, lpRasEntry);
+    lpRasEntry = NULL;
+    return ERROR_SUCCESS;
+  }
+
+  return dwRet;
+}
+
+// https://docs.microsoft.com/en-us/windows/win32/api/ras/nf-ras-rasenumentriesa
+int PrintEntries() {
+  DWORD dwCb = 0;
+  DWORD dwRet = ERROR_SUCCESS;
+  DWORD dwEntries = 0;
+  LPRASENTRYNAME lpRasEntryName = NULL;
+
+  // Call RasEnumEntries with lpRasEntryName = NULL. dwCb is returned with the
+  // required buffer size and a return code of ERROR_BUFFER_TOO_SMALL
+  dwRet = RasEnumEntries(NULL, NULL, lpRasEntryName, &dwCb, &dwEntries);
+  if (dwRet == ERROR_BUFFER_TOO_SMALL) {
+    // Allocate the memory needed for the array of RAS entry names.
+    lpRasEntryName =
+        (LPRASENTRYNAME)HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, dwCb);
+    if (lpRasEntryName == NULL) {
+      LOG(ERROR) << "HeapAlloc failed!";
+      return 0;
+    }
+    // The first RASENTRYNAME structure in the array must contain the structure
+    // size
+    lpRasEntryName[0].dwSize = sizeof(RASENTRYNAME);
+
+    // Call RasEnumEntries to enumerate all RAS entry names
+    dwRet = RasEnumEntries(NULL, NULL, lpRasEntryName, &dwCb, &dwEntries);
+
+    // If successful, print the RAS entry names
+    if (ERROR_SUCCESS == dwRet) {
+      wprintf(L"The following RAS entry names were found:\n");
+      for (DWORD i = 0; i < dwEntries; i++) {
+        wprintf(L"%s\n", lpRasEntryName[i].szEntryName);
+        dwRet = PrintEntryDetails(lpRasEntryName[i].szEntryName);
+      }
+    }
+    // Deallocate memory for the connection buffer
+    HeapFree(GetProcessHeap(), 0, lpRasEntryName);
+    lpRasEntryName = NULL;
+    return ERROR_SUCCESS;
+  }
+
+  // There was either a problem with RAS or there are RAS entry names to
+  // enumerate
+  if (dwEntries >= 1) {
+    wprintf(L"The operation failed to acquire the buffer size.\n\n");
+  } else {
+    wprintf(L"There were no RAS entry names found:.\n\n");
+  }
+
+  return dwRet;
+}
+
+void copy_dword_bytes(BYTE* bytes, DWORD value) {
+  if (bytes) {
+    union {
+      DWORD value;
+      BYTE bytes[4];
+    } converter;
+    converter.value = value;
+    memcpy(bytes, converter.bytes, 4);
+  }
+}
+
+void Demo() {
```

Comment:
Definitely cool to have this in here to show how the "magic number" is made 😄 Maybe we can put a comment over `Demo` that just explains:
```
// NOTE: This code is never called, but this is how the magic number set in
// brave_vpn::CreateEntry() is created for the `CustomIPSecPolicies` field
```

Reply:
We could also add a command line arg for this which outputs the INI output actually 😄